src/Ellipsis.js

```diff
@@ -0,0 +1,84 @@
+import React, { Component, createRef } from 'react';
+import PropTypes from 'prop-types';
+import classNames from 'classnames';
+import { throttle } from 'lodash';
+
+class Ellipsis extends Component {
+    containerRef = createRef();
+
+    state = {
+        isPrepared: false,
+    };
+
+    constructor(props) {
+        super(props);
+
+        this.throttleHandleResize = throttle(this.handleResize, 250);
+    }
+
+    componentDidMount() {
+        window.addEventListener('resize', this.throttleHandleResize, { passive: true });
+
+        this.ellipsize();
+
+        this.setState({ isPrepared: true });
+    }
+
+    componentDidUpdate() {
+        this.ellipsize();
+    }
+
+    componentWillUnmount() {
+        window.removeEventListener('resize', this.throttleHandleResize, { passive: true });
+    }
+
+    render() {
+        const { children, tag: Tag, className } = this.props;
+        const { isPrepared } = this.state;
+
+        const containerClassName = classNames('react-ellipsis_container', {
+            reveal: isPrepared,
+        }, className);
+
+        return (
+            <Tag
+                ref={ this.containerRef }
+                className={ containerClassName }
+                { ...this.props }>
```

Comment:
Won't the `className` be overriden by the `this.props` one?

We have 2 options here:
```js
const { children, tag: Tag, className, ...rest } = this.props;
...
return (
    <Tag
        ref={ this.containerRef }
        className={ containerClassName }
        { ...rest }>
``` 

or 

```js
return (
    <Tag
        { ...this.props }
        ref={ this.containerRef }
        className={ containerClassName }>
```